### PR TITLE
Dashing

### PR DESCRIPTION
**Pull request by alesolano**
Make master branch work with dashing.  
Updated joint publishers and subscribers and action client for the gripper.

Next steps:
1. update gripper subscriber
2. ~~review action client feedback callback (explained [here](https://github.com/AcutronicRobotics/mara_examples/pull/2#issuecomment-500921885))~~
3. try action client with python

### CODE REVIEW

**Comment by alesolano** [comment]
Needed to comment [these lines](https://github.com/AcutronicRobotics/mara_examples/blob/42a4b1da5c9e8cff2b7f661b5f0dbcc379b2a095/mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp#L29-L43) in the feedback callback.  
Also, I changed the `spin_until_future_complete` to `spin` [here](https://github.com/AcutronicRobotics/mara_examples/blob/42a4b1da5c9e8cff2b7f661b5f0dbcc379b2a095/mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp#L153-L161).

Now everything works, though we are not printing all the possible information (desired position, error position...).

tut_2_7_0

**Comment by alesolano** [comment]
I'll leave points 1 and 3 [here](https://github.com/AcutronicRobotics/mara_examples/pull/3#issue-287396457) for the future. Not priority now.

File: gripper_minimal_service/src/gripper_minimal_client_v2.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ class MinimalClient : public rclcpp::Node
     MinimalClient(): Node("gripper_minimal_client")
     {
       client = this->create_client<hrim_actuator_gripper_srvs::srv::ControlFinger>(
-        "/hrim_actuation_gripper_000000000004/goal");
+        "/hrim_actuator_gripper_000000000004/fingercontrol");
```
Comment (ahcorde) [inline gripper_minimal_service/src/gripper_minimal_client_v2.cpp:13]: According to the documentation this is wrong https://acutronicrobotics.com/docs/technology/hrim/model/models/actuator/gripper

File: gripper_minimal_service/scripts/gripper_minimal_client_v2.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def __init__(self):
         super().__init__('mara_minimal_client')
 
         # Create a client for service "/hrim_actuation_gripper_000000000004/goal"
-        self.client = self.create_client(ControlFinger, "/hrim_actuation_gripper_000000000004/goal")
+        self.client = self.create_client(ControlFinger, "/hrim_actuator_gripper_000000000004/fingercontrol")
```
Comment (ahcorde) [inline gripper_minimal_service/scripts/gripper_minimal_client_v2.py:14]: According to the documentation this is wrong https://acutronicrobotics.com/docs/technology/hrim/model/models/actuator/gripper

File: mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp [hunk 3]
```diff
@@ -1,151 +1,167 @@
...
-  RCLCPP_INFO(g_node->get_logger(), "Error degrees: %.2f\ttime:%.5f",
-                                        feedback->error.positions[0]*180/3.1416,
-                                        time_from_start_error);
-  RCLCPP_INFO(g_node->get_logger(), "-----------------------------");
+
+  RCLCPP_INFO(node->get_logger(), "Current degrees: %.2f\ttime:%.5f",
+                                  (double)feedback->actual.positions[0]*180/3.1416,
+                                  time_from_start);
+
+  // TODO: Not working
```
Comment (ahcorde) [inline mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp]: please decide if you are going to print it or not. If no please remove the comments

File: mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp [hunk 4]
```diff
@@ -1,151 +1,167 @@
...
-  {
-    RCLCPP_ERROR(g_node->get_logger(), "get result call failed :(");
-    return 1;
-  }
+  //
+  // ----------- GOAL SENT ----------- //
+  //
 
-  rclcpp_action::ClientGoalHandle<hrim_actuator_rotaryservo_actions::action::GoalJointTrajectory>::Result result = result_future.get();
+  // TODO: This skips the feedback and results callbacks. Response callback is fine.
```
Comment (ahcorde) [inline mara_minimal_trajectory_action/src/mara_minimal_trajectory_action.cpp]: Remove

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: gripper_minimal_service/src/gripper_minimal_client_v2.cpp (hunk 1 above)
Comment (alesolano) [reply gripper_minimal_service/src/gripper_minimal_client_v2.cpp:13]: True, we need to update the docs. `fingercontrol` is the new name

**Review by ahcorde — APPROVED** [review]

**Pull request merged**